Ginger/GingerTest/APIModelLib/POMsTest.cs

```diff
@@ -0,0 +1,215 @@
+        }
+
+
+        [ClassCleanup]
+        public static void ClassCleanup()
+        {
+            GingerAutomator.EndSession();
+        }
+
+        [TestMethod]
+        public void ValidatePOMWasAddedToPOMsTree()
+        {
+            //Act
+            mPOMsPOM.SelectPOM(mLearnedPOM.Name);
+            ApplicationPOMModel treePOM = (from x in WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ApplicationPOMModel>() where x.Name == mLearnedPOM.Name select x).SingleOrDefault();
+
+            //Assert
+            Assert.AreEqual(mLearnedPOM.Name, treePOM.Name, "POM.Name is same");
+        }
+
+
+        [TestMethod]
+        public void ValidatePOMGeneralDetails()
+        {
+            //Assert
+            Assert.AreEqual(mLearnedPOM.Name, "MyNewPOM", "POM.Name check");
+            Assert.AreEqual(mLearnedPOM.Description, "MyDescription", "POM.Description check");
+            Assert.IsTrue(mLearnedPOM.PageURL.EndsWith("TestResources/HTML/HTMLControls.html"), "POM.URL check");
+            Assert.AreEqual(mLearnedPOM.TargetApplicationKey.ToString(), "MyWebApp~843fb8a6-a844-45e9-b7ef-b045b6433c44", "POM.TargetApplicationKey is same");
+        }
+
+
+        [TestMethod]
+        public void ValidatePOMScreenshotWasTaken()
+        {
+            //Act
+            BitmapSource source = Ginger.General.GetImageStream(Ginger.General.Base64StringToImage(mLearnedPOM.ScreenShotImage.ToString()));
+            //Assert  
+            Assert.IsNotNull(source, "POM.ScreenShotImage converted to sourse check");
+        }
+
+
+        [TestMethod]
+        public void ValidateLearnedItems()
+        {
+            //Act
+            ElementInfo EI1 = mLearnedPOM.MappedUIElements.Where(x => x.ElementName == "Mexico INPUT.RADIO" && x.ElementTypeEnum == eElementType.RadioButton).FirstOrDefault();
+            ElementInfo EI2 = mLearnedPOM.MappedUIElements.Where(x => x.ElementName == "id123 input" && x.ElementTypeEnum == eElementType.TextBox).FirstOrDefault();
+
+            //Assert  
+            Assert.AreEqual(mLearnedPOM.MappedUIElements.Count, 27, "POM.MappedUIElements.Coun check");
+            Assert.AreEqual(mLearnedPOM.UnMappedUIElements.Count, 92, "POM.UnMappedUIElements.Count check");
+            Assert.IsNotNull(EI1, "POM.Element learned check");
+            Assert.IsNotNull(EI2, "POM.Element learned check");
+        }
+
+        [TestMethod]
+        public void ValidateElementsProperties()
+        {
+            //Act  
+            ElementInfo ButtonEI = mLearnedPOM.MappedUIElements.Where(x => x.ElementTypeEnum == eElementType.Button).FirstOrDefault();
+            ElementInfo ComboBoxEI = mLearnedPOM.MappedUIElements.Where(x => x.ElementTypeEnum == eElementType.ComboBox).FirstOrDefault();
+
+            //Assert  
+            Assert.AreEqual(ButtonEI.Properties.Count, 13, "POM.properties check");
+            Assert.AreEqual(ButtonEI.Properties[0].Name, "Platform Element Type", "POM.property 0 check");
```

Comment:
coverage is good now, but we should not do Index based testing- this is very breakable and will require annoying maintainance (any change will require updating all index numbers), instead, add small function to test class like "IsPropertyExist(string PropName, string PropValue) which will do Linq check and will return true/false if the property exist on no